src/connection/builders/builder_native.rs

```diff
@@ -306,4 +318,66 @@ impl<A> NativeConnectionBuilder<LinkageNotConfigured, A> {
         self.lib_path = Some(lib_path.into());
         self.safe_transmute()
     }
+
+    /// Setup the connection using the string
+    /// pattern.
+    ///
+    /// Basic string format: `firebird://{user}:{pass}@{host}:{port}/{db_name}?charset={charset}&lib={fbclient}&dialect={dialect}`
+    ///
+    /// Some considerations:
+    /// - If you not provide the host, we will consider this a embedded connection.
+    /// - The port, user and pass parameters only will be accepted if you provide the host.
+    /// - If you not provide the `lib={fbclient}`, we will consider this a dynamic linked connection. The default, by the way.
+    pub fn with_string<S: Into<String>>(
+        self,
+        s_conn: S,
+    ) -> Result<NativeConnectionBuilder<DynByString, ConnByString>, FbError> {
```

Comment:
There is no need to own the string.

```suggestion
    pub fn with_string(
        self,
        s_conn: &str,
    ) -> Result<NativeConnectionBuilder<DynByString, ConnByString>, FbError> {
```

Reply:
Also, i think the name `from_string` would be better, as we are constructing a connection from a connection string.